### ISSUE TITLE
[Dev Asks] Solar System Size Reduction

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#4666]
So I have been doing a lot of tech lately.
One point I need to tackle for some time is the shrinking of the solar system.
It was needed for gameplay reason, but after further time spent on the tech, having a smaller world would also be a massive plus if I can get it below a certain threshold.

It would means:
- I can ditch my second camera used to render far away stuff.
- Very large brick entity show far away, with decals. 
In short it makes my life much much easier (normal decals would simply be impossible btw with dual cam), but would also be faster on the CPU/GPU side as I dont need to run some things twice. Also having a brick entity instead of the torus would lead massive performance gains (+20/25% fps on a calm scene).
It would put also less stress on upcoming tech like asteroids and planets.

Now what size are we talking about?
Do to floating point issue I would have a camera far plane of 5000km, so that is the world size I have to play with.
That would mean roughly a 10x division: Sun would be 200km in diameter. Gas planet would be 100km. A normal planet would be 50km (those are average, some would be bigger). I would have to bunch things up slightly closer too. Granted those are not large number but they are not small either, standing on a 50km planet will feel alright (it does in No Man Sky). 
Compared to other game like No Man Sky or Space Engineers it seems equivalent or a bit below. But still bigger than Avorion ^^.
I could do like NMS which does not use km but its own unit, so the player dont really get things are not as large as real life stuff.
For gameplay reason I intended to do a 5x division anyway, it will make navigation and a ton of other stuff easier.

I tried in-game and it is of course it is quite a different feel but I think that would totally work. Some stuff looks better like the asteroid  ring. I would have to reduce the travel speed of course.

Tell me what you guys think! I still need to run further testing to see if I missed some more issues or not, I am still not confident single cam can work at this stage. I can fine tune those numbers too.
Cheers

**Comment by Kaiser-Indrasil** [comment]
"Still bigger than Avorion's", that's a good one XD

So yeah, there's obviously great things to gain with this transition, as you've explained.

It works for Space Engineers, you're right about that. Their planets are 120 km in diameter at maximum, but it's a bit too easy to see the curvature of the planet from its surface, so maybe a 1/5 scale would be possible with the planets being slightly closer together?

With the downscale, it would be way easier to find people in multiplayer servers, once they are implemented, so that creates more opportunities for co-op and PvP.

There's obviously the downside of losing the sense of scale that for example, the big-ass gas giant provides.
That's why I'm advocating for a 5x downscale, to keep the planets less Outer Wilds-ish (astounding game btw) and more Star Citizen-ish.

All in all, if the downscale means we'd be able to see the blocks across way longer distances, and the sticker rendering distance was improved, I'm all aboard this change.

**Comment by PFSCommissar** [comment]
The sheer scale of things is something that makes this game fun to play. I’ve never been in a space sandbox that attempts the scale SEVO does and it’s beautiful. But at the same time i already feel thrown off by knowing that the planets are much closer than they are in real life. I would hate to see the scale reduced even more and i feel it would make the game feel less serious. 

What if you tried something else, then? Would it be possible to space things out a bit further, and use super low res planet models at that distance, since they’d just be specs of light anyway? 

Alternatively, if you plan to one day add planet terrain, smaller planets might work. It did work in SE, though it was certainly weird at times, so i’d agree with Kaiser’s 5x suggestion in that case. 

However, seeing planets close together has been something that has always broken my immersion very quickly and IMO ruins the seriousness of the game and looks terrible. You could take KSP for an example. The planets are small and a bit dorky, but they’re still very far apart which brings some seriousness back to the game. So if you plan to scale down solar systems, i’d also scale down the number of planets. But overall i’d just really hate to see the bodies closer.

**Comment by schlid** [comment]
Decals showing from far away would be great. In the same boat as indrasil here, there are obvious benefits to making the universe a little smaller. It seems like a "nice to have" if the scale could be preserved but in multiplayer there would be so much barren space, and there still will be even with a down scale. 5000km is massive.

**Comment by Kaiser-Indrasil** [comment]
I do agree that setting the planets further apart adds to the realism aspect and makes the setting more believable.
That said, I also believe that bringing them closer together can really improve the beauty of the scene.
Both approaches have their pros and cons and could work with the game either way.
In that case, when we are limited by the technology, it would make more sense to bring them together if that means we could end up with 5x shrink instead of the 10x one.

**Comment by PFSCommissar** [comment]
If done well, lowering planet scale would offset the closer proximity visually. I definitely would say they shouldn’t *appear* closer than they do now. I’d also say that there should be fewer gas giants, and make some even smaller planets to be moons of the terrestrial ones instead. Tbh, flying into some systems and seeing two ringed giants in close proximity is a bit cringey, but then i’m also an aerospace engineer so physics is in my blood now.

**Comment by ProPeach** [comment]
This is a real interesting one, thanks for opening up the discussion o/
For my part, I think I'm pretty neutral on this. The benefits you talk about sound fantastic, especially the performance and decal render distance elements. However, ~50km planets do seem quite small to me. In SE, their 120km planets feel great when you're directly on the surface, although once you fly up even 100m or so the illusion is very quickly broken and you see how even each individual mountain has a curve to it. At half the scale, this effect would be even more pronounced. Perhaps the feeling could be reduced somewhat by some pretty atmospheric shaders, it's hard to tell speculatively. Ideally I would prefer to stick around this 120km mark at least for the "standard" planet, not lower if that can be avoided.
Starship EVO has always had a fantastic feeling of scale, helped a lot by the scale of building - it looks great at the human scale, and the galactic scale of megastructures. I feel like bringing things closer will reduce that feeling a little so as others have said the KSP approach of smaller planets are larger distance scales would be my preference.
Looking forward to seeing how this goes ahead!

**Comment by PFSCommissar** [comment]
> This is a real interesting one, thanks for opening up the discussion o/
> For my part, I think I'm pretty neutral on this. The benefits you talk about sound fantastic, especially the performance and decal render distance elements. However, ~50km planets do seem quite small to me. In SE, their 120km planets feel great when you're directly on the surface, although once you fly up even 100m or so the illusion is very quickly broken and you see how even each individual hill has a distinct curve to it. At half that scale for 50km, this effect would be even more pronounced. Perhaps the feeling could be reduced somewhat by some pretty atmospheric shaders, it's hard to tell speculatively. Ideally I would prefer to stick around this 120km mark at least for the "standard" planet, not lower if that can be avoided.
> Starship EVO has always had a fantastic feeling of scale, helped a lot by the scale of building - it looks great at the human scale, and the galactic scale of megastructures. I feel like bringing things closer will reduce that feeling a little so as others have said the KSP approach of smaller planets are larger distance scales would be my preference.
> Looking forward to seeing how this goes ahead!

Agreed. Going smaller than SE would be a mistake, especially if you have plans for terrain in the future

**Comment by kynamats** [comment]
The technical upsides sound really major, but I have to agree with others that this could risk making planets feel too small in future. To counteract this I would be ok with NMS style larger planets; even though its less realistic I feel it contributes well to the space opera vibe, and can make some really incredible vistas.

**Comment by ProPeach** [comment]
I've done a bit more thinking and looked at some NMS gameplay. Previously I've only played around with Space Engineers, and as I mentioned their planets do feel quite small and "fake". Despite being half the scale in absolute terms however, the planets from NMS appear much more convincing and the terrain looks flatter at altitude. Considering this if you can pull off the same desception here then I would be OK with the smaller ~50km size I think, however it does seem like there is a lot more work involved in hiding that fact like improved atmospheric VFX and perhaps a distortion effect to make the terrain seem flatter when you fly down to it.  In NMS the transition between space and planet flight is also much harsher which in sure hides a lot of these tricks, it almost feels like a loading screen compared to the far smoother and more transparent transition of SE.

So I suppose it depends on where you want to spend effort, these additional atmospheric and distortion effects seem to be really worth it, but a small 50km planet without them would feel quite toy-like which I wouldn't enjoy personally

**Comment by Camzx** [comment]
Less is more sometimes. You could probably change it again later, right?
And I'm assuming it being closer wont impact performance.

**Comment by TIKIRobo** [comment]
If its not too difficult you could always make a build or a video showing off planets of different scales (super basic planets ofc) that way players can see the difference between a 50km, a100km and a 150km planet and see how it curves over the horizon and looks when you're flying in 

Making a video might be much easier but some people might want to see them in game for themselves

**Comment by Uncle-Ulty** [comment]
The Planet size is a super critic decision that influences gameplay immersion and performance. Tbh, I can't decide about it based on speculations, parameters, and numbers... IMO the best way to decide would have some explorable planets of different sizes where we could land, build and interact, and so make up our mind. I believe the following diameters would be perfect for the tests: 500 , 250, 100, 75, 50, 25.

I know it would be a lot of work to make something that would be prbly discarded, but I believe the planet's size is a very critical decision to make.

But, if it's not possible to make some variety of sizes for testing purposes, and the 50km size is already settled one, let me ask something. Back in 1999, a game called "Silent Hill" was released into PS1. This game used a fog effect on the exteriors zones cuz it helped to hide the far distance and not rendered objects in the map. It helped the game performance, since it was a PS1 game. Thus, let me ask something, if the 50km size planet is already decided, would it be possible to have bigger but foggy planets? I mean, using the same strategy back in 1999 Silent hill, would it be able to spawn big planets (75 km, 100 km, and so on) with fog effect, to reduce the rendering requirement? A volcanic planet with lots of gas would have a fog of vapour and suspended dust. A planet with a more dense vegetation would have lots of mist. A desert planet would have lots of sand storms effect. All this would reduce the rendering impact and allow a bigger planet, or am I wrong?

These are some ideas, cuz I think bigger vs small planets have lots of pro and cons for the gameplay immersion and the computational performance.

**Comment by ZachZent** [comment]
I got a possible workaround which will allow the lowering of the system size while keeping the planets semi-distant. Now based on old conversations you did not want planets revolving around the sun, which makes sense (though you did want rotate for a day/night cycle). Simply set a upper limit of number of planets in a system and make sure they are spaced out so no two planets are within X km from each other. Then slow down the cruise speed. That way planets will still look far enough away from each other and the trip between them will no seem like a walk across the street.

You can also add some "fog of war" to help disguise it further. Right now everything is bright and clear, which is good. But some classic sci-fi clouds in the system to fog it up a little. Then dim the planets visual profile from further away so it can't be easily seen from a near by planet. There are a lot of tricks to disguise the reduction in scale

![image](https://user-images.githubusercontent.com/36980446/161391740-1447a577-4a37-44bf-bdc9-1ee8ea45a779.png)

**Comment by FlyingDebris** [comment]
> I got a possible workaround which will allow the lowering of the system size while keeping the planets semi-distant. Now based on old conversations you did not want planets revolving around the sun, which makes sense (though you did want rotate for a day/night cycle). Simply set a upper limit of number of planets in a system and make sure they are spaced out so no two planets are within X km from each other. Then slow down the cruise speed. That way planets will still look far enough away from each other and the trip between them will no seem like a walk across the street.
> 
> You can also add some "fog of war" to help disguise it further. Right now everything is bright and clear, which is good. But some classic sci-fi clouds in the system to fog it up a little. Then dim the planets visual profile from further away so it can't be easily seen from a near by planet. There are a lot of tricks to disguise the reduction in scale
> 
> ![image](https://user-images.githubusercontent.com/36980446/161391740-1447a577-4a37-44bf-bdc9-1ee8ea45a779.png)

Adding on to this, in astrophotography you'll experience a significant lens flare when looking at an object like a planet from far away, especially one where you can see the daytime side of it. It could be entirely possible to cover a larger planet with some sort of feature that gives the appearance of being much farther away as well as obscuring the planet itself from view:

![image](https://user-images.githubusercontent.com/38392651/161392545-1cdcdb2b-fe2f-4c0d-b392-05b32b74b251.png)


Something like this could scale for big or small planets, but has a potential downside of needing to find a good balance between that cutoff point where the planet becomes more visible than the sprite or whatever effect ends up being used. A workaround for that could be to expand the size of the sprite until the fake "planet" in the center matches the angular size of the real one?

**Comment by pinesh** [comment]
Scaling down this stuff is absolutely fine. Players will always light cruise to their destinations anyway so it will hardly feel like the universe is drastically smaller.

**Comment by Tisdag1** [comment]
Personally I am opposed to shrinking the scale down yet again. The incredibly massive & awe inspiring scale is the biggest (no pun intended) thing that drew me to this game when I first came across it years ago. Imho everything is already starting to feel too small & cramped (especially with how fast we can go). Everything has been shrunk down a few times over the years already, and the planets & systems are way smaller now than when I backed the game years ago.

 And now you are wanting to shrink it down even smaller? And what's worse is, you are wanting to make the planets 2.5x smaller than the standard SE planets so that even our "Gas Giants" are smaller than the standard SE planets? The 120km size of their normal planets are already way too small in my opinion. And the curvature of the planets is so glaringly noticeable from just a few meters off the ground that it feels like I am playing on, and flying around a larger version of the planets from the games Spore, or outer wilds.

I am sorry for the rant. Like I said, the scale is the biggest thing that interested me when I backed the game. So to see the game lose that and everything become smaller than a game that is regularly mocked on the Discord channel is disheartening. Unfortunately I don't really have anything constructive to add & I am not sure what you can do to help minimize the impact of it. If it's something you need to do, then you need to do it & I will just need to suck it up and deal with it. lol

**Comment by pinesh** [comment]
Large planets are something we just don't need. Player reception to the ringworlds has been largely positive, and they're just a few dozen KM across. There's no incentive to commit to such large worlds if there won't be anything interesting in them. Keep it tight, keep it focused and keep it concise.  There's a massive galaxy already present. If everyone could fit on the one planet, why would we seek out the stars?

**Comment by ZachZent** [comment]
An old post I made for planet generation. Even with the small size, unique but structured planets can still be a thing

https://github.com/tsunamayo/Starship-EVO/issues/2061

**Comment by Briaireous** [comment]
I am in support of the size reduction of the system and planets for higher performance. However, I would like to know if you would consider increasing their size in the future if the tech for better performing large worlds is avaliable?

**Comment by TwoNiner29** [comment]
I agree I think planets don’t have to be massive for gameplay to be fun. The only thing we would lose is immersion. I think indrasil’s idea of scaling them down 5X is a good balance.

**Comment by Arutus** [comment]
I disagree with having planets at 50km, they should at the very least be 75-100km with such a small size you could hardly utilize anything big on it as you'd travel the planet far too quickly, the scale of this game is what keeps me here, to take that away ruins it

**Comment by Regnion** [comment]
Having played Space Engineers with planets extensively, having smaller planets will not hurt this game.  There will still be plenty of room.

The more important thing is maintaining visual consistency, in that the planets still appear to be far apart as opposed to lumped closely together.  I also recommend few ringed gas giants, even just 1 per system.  Perhaps also consider making a smaller ring variant, like Saturn's stunning rings vs. Neptune's very faint and thin rings.  You'll want to keep things varied and visually interesting.

**Comment by PFSCommissar** [comment]
I think planets and solar systems have to be visually interesting, otherwise they’ll be disappointing to explore. A big part of that is scale. Lowering scale will lower the seriousness of the game, which can also reflect negatively on creations and make serious ones a bit silly by nature of their context. Tbh i’ve never been a fan of the characters in this game as i feel like they take away that seriousness, and i’vee seen negative reviews of sevo where people feel it’s childish. Granted thats a very surface level review and that person certainly needs to do more research, but i do worry about the game feeling very silly when planets get too small

**Comment by Kaiser-Indrasil** [comment]
What you need to understand, is Stevo has never been and was never meant to be a sim. It's always been an arcade game at it's heart. Therefore, it has no business being all serious. Convincing, consistent, and fun, yes, but not a simulation. There's EVE, there's KSP, there's E:D, but they lean towards realism more than Stevo ever meant to. You will still be able to build realistic looking spacecraft with functional mechanics like fuel and the like but in smaller solar systems. Not like this matters that much, since we've got a whole galaxy to explore and fight over anyway.

**Comment by PFSCommissar** [comment]
Respectfully, i don't think that's a statement you get to make. SEVO is a constantly evolving game, and we're part of shaping it. that's why Tsuna is asking for our opinion here. Some of us are going to want it to be more serious than others are and there's nothing wrong with that. 

I don't want to play EVE because the community is a bunch of scammers, KSP is great but you can't make beautiful ships like you can in SEVO, and E:D and EVE you can't make ships at all. I come to SEVO because i've found it to be the best space sandbox so far. SE was too realistic for its own good, where child entities went crazy and you could never dream of the beautiful moving parts, concealed weapons, and landing gear that SEVO offers. That's what convinced me to leave the more realistic SE for this. I can accept SEVO's less serious atmosphere for what it offers, but there could come a point where honestly it doesn't feel worth it anymore because of a silly looking context. I definitely don't feel like we're there, and i may very well not feel like that whenever planets shrink, but there's nothing wrong with offering caution against heading in that general direction.

**Comment by Kaiser-Indrasil** [comment]
I can get behind that. That said, I do see where tsuna is coming from with this proposed change.

With all that, we cannot really tell if it was a good change without actually being able to set foot on a planet and seeing the changes from its surface and how it ties to the rest of the game.

I think planetary tech should come first for the planet size that we currently have and there should be moons included which would reflect the size of the proposed shrinked planets, so that we may make a more educated decision.

**Comment by PFSCommissar** [comment]
Agreed. That would certainly would be a better order to do things in. And if we then find it's possible to the NMS deception that others have mentioned, that could be a great compromise.

**Comment by Horkas** [comment]
Having a smaller more "condensed" game is the best actions you can take. One of the biggest weaknesses of space games is that space is filled with.... Well space. Nothingness. That's boring. What makes space fun is what you DO in space. Compacting things to be closer and more dense will make the game feel allot more fun overall. Flying around needlessly huge planets and space filled with nothing will not win you any favor from the players.

**Comment by Regnion** [comment]
> That would mean roughly a 10x division: Sun would be 200km in diameter. Gas planet would be 100km. A normal planet would be 50km (those are average, some would be bigger). I would have to bunch things up slightly closer too. Granted those are not large number but they are not small either, standing on a 50km planet will feel alright (it does in No Man Sky).

After giving this some more thought, I thought it would be good to point out something:

**Don't let your drive to create a very efficient game undermine your need to create an enjoyable game.** Obviously they're not mutually exclusive, obviously a game needs to run to be played and be enjoyed. What I'm saying is, don't sacrifice a quality and immersive experience for the sake of min-maxing those numbers.

I feel that having large planets is a good thing, and that the feeling of actually being in space with gas  giants and a tremendous star burning in the center is amazing; I also understand the need to make things more efficient and accessible to more people who don't have rigs as quality as some.

I would suggest you consider a middle ground, a 5x division instead of 10x. Will that get you to your ideal target? Perhaps not. Will the planets be as large as people want? Also no. However, you get to stand on a large planet with a barely noticeable curvature. You still get that awesome scale that you enjoy when flying around in lightcruise. There is still a feeling if grandeur and vastness to space, and it will still be an improvement in efficiency.

The middle ground is sometimes summed up as "nobody is happy," but really it's "everyone is acceptably pleased." You get an efficiency boost to gameplay, and you get large planets and systems that still feel epic.

**Comment by ZachZent** [comment]
I don't mind the reduction AS LONG AS the planets are large enough that you can't tell the difference and 50k planets are large enough for that effect. Aka the horizon is far enough away that you can't see the planet curve under you. Moons assuming semi-equivalent size would be a bit more odd as they may be small enough that the horizon would be far more obvious or large enough that it looks very strange next to the planet.

However I am concerned about the overall system size. Even with the reduction in planets, the overall size does seem overly small. Even with slowing down cruise (and slower flight speeds in atmo), lowering planet luminosity, and other tricks I mentioned above, you still have to consider gas giants and any space phenomenon that may be added. 
- As an aside, I don't think gas giants and stars need to be massively scaled down since stars do not need to have any physical substance besides the model/animation (damage effects doesn't need to be tied to the model, just the position) and gas giants while do need some substance for say 10km down before instant death, it isn't physical, breakable substance.

With stars and the system being that small, even with slow speeds, everything is going to fell very tight both visually and practically. Even with the above trick of spacing them out, it will feel a lot more like astroneer with its micro-system. The system doesn't need to be large, but it needs to be spaced (hahaha)

**Comment by pinesh** [comment]
> **Don't let your drive to create a very efficient game undermine your need to create an enjoyable game.** Obviously they're not mutually exclusive, obviously a game needs to run to be played and be enjoyed. What I'm saying is, don't sacrifice a quality and immersive experience for the sake of min-maxing those numbers.
> 

It's not so much about efficiency, It's also to do with the fundamental limitations of what's possible in Unity.
> Do to floating point issue I would have a camera far plane of 5000km, so that is the world size I have to play with.

**This** is a fundamental part of Unity. If we want these rendering features, if we want that "immersive" experience, we have to accept limitations to system sizes. Those 10x reductions are proportional to the needed reduction in world size. A 5x would lead to a much tighter system squeeze.

Multi kilometer vessels have never been a core game design choice going back as far as Kickstarter. There's been a number of community posts that mockup 50km planets, and you know what, they look great. Sure it's a bit more arcadey, but it's 100% in style with the game. Changing the measuring units and speed will all make the transition easier. 

A 5000km system will take almost 5 hours to cross in the game's current build at sub-light cruise speeds. Since nobody in their right mind would do that, Size is arbitrarily linked to LC/warp speed. Everything can be put in proportion.

There are 1.1 **million** Star systems in the Stevo galaxy. That's more room than anyone needs.

//Courtesy of Smets
https://imgur.com/a/8UxF2bX
//Courtesy of Xerma
![image](https://user-images.githubusercontent.com/14081996/161845311-c3200b23-05be-4d48-acf8-53f00755c806.png)

//A venator 1.7km from the ground
![image](https://user-images.githubusercontent.com/14081996/161845441-4a78eb02-7162-4ae4-b3d1-e122f875b6f7.png)

**Comment by ExodistSKY1** [comment]
As much as I wish to keep the size of the game that it is currently due to loving the sheer size of the game when compared to others, this made it stand out. Flying through the ring at light speed made you feel the awww that this game has done so well with doing.  IMO 50km moons would be okay but I would still want to try to keep the awww that this game has and have 150km planets. I understand both arguments, it seems that this game has just continuously scaled down time and time again, I am not making a jab it is just something that was vastly different than other games in a great way. If we can try to keep this by not scaling down as far that would be ideal. But its understood that things do need to be adjusted.  

I don't know if adjusting speed is the solution, I love that we can get upwards of, now after the recent change, to 400m/s which I think is a good shift speed for smaller craft.  Changing the name of m/s to something else wouldn't make it seem any slower or faster.  Would the base player speed also be changed because that would be weird if you can run half as fast as a ship or vehicle. 

I do have a question regarding the,  "floating point issue I would have a camera far plane of 5000km, so that is the world size I have to play with" 

Is this so that everything is visible at a single time within each solar system? and is that the reason why the 5000km limit is in place?

Like I stated above, I truly love this game and want what is best for it, but other than the building mechanics, which are awesome and best of the market, the vastness and size of the solar systems is what grasps at me and I think I can safely say others

**Comment by tsunamayo** [comment]
Thanks a lot everybody! As expected people wants different things for the game, so I wont be able to satisfy everybody.
I too want a somewhat immersive experience but not at the expense of gameplay. Immersive is also very different from "real" or "authentic", as real scale is actually feels less immersive in a sense (like in Elite, where supercruise feels and is completely disconnected from normal gameplay, but that is how is should be given the scale difference).
It was long and painful but I have been making progress, so I will keep a dual camera setup with a few tricks. "Close" camera far plane will be at 1000km (instead of 50km currently). But brick entity can renderer on the far cam (just not the decals), so I will be able to replace the space habitat.
I managed to improve performance so we should not see any significant perf impact versus what we have now (I still pay a price because of the dual cam though).
=> this means I can choose my scale freely. So likely 100km-ish planets (which they are currently btw, but I will reduce sun and gas planets to get a smaller system), I just want to do some more test before I decide on something.
Thanks
